core/sequencer/server.go

```diff
@@ -323,21 +323,22 @@ func (s *Server) ApplyRevisions(ctx context.Context, in *spb.ApplyRevisionsReque
 		return nil, err
 	}
 
-	for i := int64(1); i < int64(s.LogPublishBatchSize); i++ {
+	i := int64(1)
+	for ; i <= int64(s.LogPublishBatchSize); i++ {
 		req := &spb.ApplyRevisionRequest{
 			DirectoryId: in.DirectoryId,
 			Revision:    highestApplied + i,
 		}
 		_, err := s.loopback.ApplyRevision(ctx, req)
 		if st := status.Convert(err); st.Code() == codes.NotFound {
 			unappliedRevisions.Set(0) // Outstanding revisions = 0
-			return &empty.Empty{}, nil
+			break
 		} else if err != nil {
 			return nil, err
 		}
 
 	}
-	glog.Warningf("ApplyRevisions: too many outstanding revisions; applied %d", s.LogPublishBatchSize)
+	glog.Warningf("ApplyRevisions: applied %d revision(s) [%d, %d]", i-1, highestApplied+1, highestApplied+i-1)
```

Comment:
Is it not too many warnings? Previously it was only printing it if there were too many revisions.

Also, while you are here, the log line does not mention the directory ID which makes it slightly less useful.

Reply:
Ah should be info